wiki/plugins/metadata/models.py

```diff
@@ -791,13 +821,15 @@ class CorpusSentence(models.Model):
     note = models.CharField(max_length=200, blank=True, verbose_name="Annotator Note")
     mwe_markup = models.CharField(max_length=200, blank=True, verbose_name="MWE Markup")
 
-    def get_absolute_url(self):
+    @cached_property
+    def url(self):
         return reverse('wiki:corpus_sentence_view', args=[self.language.slug, self.corpus, self.sent_id])
 
+    @cached_property
     def html(self):
-        return format_html('<a href="{}" class="corpussentence">{}</a>', self.get_absolute_url(), self.sent_id)
+        return format_html(f'<a href="{self.url}" class="corpussentence">{{}}</a>', self.sent_id)
```

Comment:
Could this line be `format_html(f'<a href="{self.url}" class="corpussentence">{self.sent_id}</a>') ` ?

Reply:
No, the difference is that `format_html` escapes the arguments before inserting them into the pattern. I believe the URL should already be properly escaped, but the sentence ID could have special characters.